File: Section 2/Nicholson_Bailey_Jury_Functions.m
%% The Mathematics of Host-Parsitoid Population Dynamics:
% This file constructs the Jury functions as functions of R for the
% Nicholson-Bailey model and plots them versus R.  The output is a graph
% that shows how the first two Jury functions are positive and the last
% Jury function is negative.  This shows that the coexistence fixed point
% of the Nicholson-Bailey is linearly unstable. 
% (This file creates the right panel of Figure 6 in Section 2.2.)
clear all, close all, clc

% Discretizing R space: 
R = linspace(1,7,1e4);

% Jury functions:
J1 = @(R) log(R);
J2 = @(R) 2 + log(R).*(R+1)./(R-1);
J3 = @(R) 1 - log(R).*R./(R-1);

% Plot: 
figure(1)
plot(R,J1(R),'k','linewidth',5)
hold on
plot(R,J2(R),'r','linewidth',5)
plot(R,J3(R),'b','linewidth',5)
set(gca,'fontsize',18)
title('Stability Analysis of N-B Model','fontsize',25,...
      'interpreter','latex')
xlabel('$R$ (viable eggs per adult host)','fontsize',22,'interpreter',...
       'latex')
ylabel('Jury Functions','fontsize',22,'interpreter','latex')
legend('Jury 1','Jury 2','Jury 3','location','Northwest','interpreter',...
       'latex','fontsize',22)
grid on
grid minor

File: Python Program Files/Section_2/NIcholson_Bailey_Jury_Functions.py
"""
%% The Mathematics of Host-Parasitoid Population Dynamics:
% This file constructs the Jury functions as functions of R for the
% Nicholson-Bailey model and plots them versus R.  The output is a graph
% that shows how the first two Jury functions are positive and the last
% Jury function is negative.  This shows that the coexistence fixed point
% of the Nicholson-Bailey is linearly unstable. 
% (This file creates the right panel of Figure 6 in Section 2.2.)
"""

# Import libraries:
import numpy as np
import matplotlib.pyplot as plt

# Discretizing R space: 
R = np.linspace(1, 7, int(1e4))

# Jury functions:
J1 = lambda R: np.log(R)
J2 = lambda R: 2 + np.log(R) * (R + 1) / (R - 1)
J3 = lambda R: 1 - np.log(R) * R / (R - 1)

# Plot: 
plt.figure(1)
plt.plot(R, J1(R), 'k', linewidth=5)
plt.plot(R, J2(R), 'r', linewidth=5)
plt.plot(R, J3(R), 'b', linewidth=5)

ax = plt.gca()
ax.tick_params(labelsize=18)
plt.title('Stability Analysis of N-B Model', fontsize=25)
plt.xlabel(r'$R$ (viable eggs per adult host)', fontsize=22)
plt.ylabel('Jury Functions', fontsize=22)
plt.legend(['Jury 1', 'Jury 2', 'Jury 3'],
           loc='upper left', fontsize=22)
plt.grid(True, which='both')
plt.minorticks_on()
plt.show()
